external/java/src/main/java/ch/epfl/dedis/byzcoin/StateChange.java

```diff
@@ -0,0 +1,101 @@
+package ch.epfl.dedis.byzcoin;
+
+import ch.epfl.dedis.lib.proto.ByzCoinProto;
+import com.google.protobuf.ByteString;
+
+/**
+ * Represents the state change of an instance and thus contains all the
+ * information related.
+ */
+public class StateChange {
+    private StateAction stateAction;
+    private ByteString instanceId;
+    private ByteString contractId;
+    private ByteString darcId;
+    private ByteString value;
```

Comment:
some of these types can be improved, I think we have an InstanceID class, contractID is usually a string in other classes, DarcID probably has its own class too.